components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/tokenprocessor/HashingPersistenceProcessor.java

```diff
@@ -104,6 +105,10 @@ public String getPreprocessedRefreshToken(String processedRefreshToken) throws I
      */
     private String hash(String plainText) throws IdentityOAuth2Exception {
 
+        if (StringUtils.isEmpty(plainText)) {
+            throw new IdentityOAuth2Exception("plainText value is null or empty to be hash.");
+        }
+
         MessageDigest messageDigest = null;
         byte[] hash = null;
         String hashAlgorithm = OAuthServerConfiguration.getInstance().getHashAlgorithm();
```

Comment:
Hashed entry we store in the database should be self contained.. ie. we need to store both hashing algorithm and along with the hashed value..

So the value we store in the database would like this.
{algo:"xxxx",hash:"yyyy"}

Reply:
Additionally as per the implantation of this class, all tokens also get hashed; not only the client secret..

It should have the  only client secret hashed, based on the new configuration introduced with this PR.. configuration..

Reply:
Fixed in [1].
[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/886/files#diff-cd6fc5f0e2c2a8f8a78e6a4317c2cf2dR127